scm-ui/ui-webapp/src/repos/permissions/containers/AdvancedPermissionsDialog.tsx

```diff
@@ -21,93 +21,80 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React from "react";
-import { WithTranslation, withTranslation } from "react-i18next";
+import React, { FC, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
 import { Button, ButtonGroup, Modal, SubmitButton } from "@scm-manager/ui-components";
 import PermissionCheckbox from "../../../permissions/components/PermissionCheckbox";
 
-type Props = WithTranslation & {
-  readOnly: boolean;
+type SelectedVerbs = { [verb: string]: boolean };
+
+type Props = {
   availableVerbs: string[];
   selectedVerbs: string[];
-  onSubmit: (p: string[]) => void;
+  readOnly: boolean;
+  onSubmit: (verbs: string[]) => void;
   onClose: () => void;
 };
 
-type State = {
-  verbs: any;
+const createPreSelection = (availableVerbs: string[], selectedVerbs?: string[]): SelectedVerbs => {
+  const verbs: SelectedVerbs = {};
+  availableVerbs.forEach(verb => (verbs[verb] = selectedVerbs ? selectedVerbs.includes(verb) : false));
+  return verbs;
 };
 
-class AdvancedPermissionsDialog extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    const verbs = {};
-    props.availableVerbs.forEach(
-      verb => (verbs[verb] = props.selectedVerbs ? props.selectedVerbs.includes(verb) : false)
-    );
-    this.state = {
-      verbs
-    };
-  }
-
-  render() {
-    const { t, onClose, readOnly } = this.props;
-    const { verbs } = this.state;
-
-    const verbSelectBoxes = Object.entries(verbs).map(e => (
-      <PermissionCheckbox
-        key={e[0]}
-        name={e[0]}
-        checked={e[1]}
-        onChange={this.handleChange}
-        disabled={readOnly}
-        role={true}
-      />
-    ));
-
-    const submitButton = !readOnly ? <SubmitButton label={t("permission.advanced.dialog.submit")} /> : null;
+const AdvancedPermissionsDialog: FC<Props> = ({ availableVerbs, selectedVerbs, readOnly, onSubmit, onClose }) => {
+  const [verbs, setVerbs] = useState<SelectedVerbs>({});
+  const [t] = useTranslation("repos");
+  useEffect(() => {
+    setVerbs(createPreSelection(availableVerbs, selectedVerbs));
+  }, [availableVerbs, selectedVerbs]);
 
-    const body = <>{verbSelectBoxes}</>;
+  const handleChange = (value: boolean, name?: string) => {
+    if (name) {
+      setVerbs({
+        ...verbs,
+        [name]: value
+      });
+    }
+  };
 
-    const footer = (
-      <form onSubmit={this.onSubmit}>
-        <ButtonGroup>
-          {submitButton}
-          <Button label={t("permission.advanced.dialog.abort")} action={onClose} />
-        </ButtonGroup>
-      </form>
-    );
+  const verbSelectBoxes = Object.entries(verbs).map(verb => (
```

Comment:
You can do .map(([name, value]) => (...))
It could be more readable than verb[0] and verb[1] I think